cloud/app/server/utils.js

```diff
@@ -47,4 +52,78 @@ const getVersionRange = (version, type) => {
   return range.format();
 };
 
-module.exports = { getNextInRange, getVersionRange };
+
+/** Ensure ClickHouse database and user exist for a capability
+ * Only used with admin ClickHouse user credentials.
+ * @param {string} capName - capability name
+ * @returns {Object} - {dbName, user, password}
+ */
+const setupCapabilityDB = async (capName) => {
+  const dbName = `cap_${capName.replace(/@/g, '').replace('/', '_').replace(/-/g, '')}`;
+  log.debug(`Setting up ClickHouse database: ${dbName}`);
+
+  await ClickHouse.client.exec({
+    query: `CREATE DATABASE IF NOT EXISTS ${dbName}`
+  });
+
+  const user = `${dbName}_user`;
+  log.debug(`ensuring clickhouse user ${user} for database ${dbName}`);
+  // Check if user exists
+  const userExists = await ClickHouse.client.query({
+    query: `SELECT name FROM system.users WHERE name = '${user}'`,
+    format: 'JSONEachRow'
+  });
+
+  const users = await userExists.json();
+  const mongoCredentialsCollection = Mongo.db.collection('clickhouse_users')
+  if (users.length > 0) {
+    // retrieve password from mongo
+    const userDoc = await mongoCredentialsCollection.findOne({ user: user, db: dbName });
+    if (userDoc) {
+      log.debug(`ClickHouse user ${user} for database ${dbName} already exists`);
+      return { dbName, user, password: userDoc.password }
+    }
+  }
+
+  // Generate new password if user doesn't exist
+  const password = Math.random().toString(36).slice(-12);
```

Comment:
Maybe use `getRandomId` from utils instead? This code here always starts with a '.' and I don't think we need to test a function like this in more than one place.